### ISSUE TITLE
 can’t establish a connection to the server at wss://profileapp.org:6001/app/appkey1?

### ISSUE THREAD

**Issue by anwarhamoude** [beyondcode/laravel-websockets#148]
I've poked at every setting, there has to be more to it- what special thing don't I know?

The location is--> https://profileapp.com

The issue is described here- might be easier to read. --> https://profileapp.org/posts/5

ERROR: can’t establish a connection to the server at wss://profileapp.org:6001/app/appkey1?protocol=7&client=js&version=4.4.0&flash=false. app.js:1:259475

ERROR: The connection to wss://profileapp.org:6001/app/appkey1?protocol=7&client=js&version=4.4.0&flash=false was interrupted while the page was loading.

Works fine on my development machine. Yeah great right. It is disappointing that it doesn't connect- you can see the error when you open the console. My question is how can this work? As I can only get so much joy with 2 browser windows open writing myself messages back and forth on my computer.

Server: Vultr.com, Ubuntu 18.04, NGINX with Certbot.

Anyone with the expertise to solve the mystery? The details are as follows.

Here is NGINX configuration: $ /etc/nginx/sites-available/default

map $http_upgrade $type {
default "web";
websocket "ws";
}

server {

root /var/www/laravel/public;

index index.php index.html index.htm index.nginx-debian.html;

server_name profileapp.org www.profileapp.org;

location / {
try_files /nonexistent @$type;
}

location @web {
# First attempt to serve request as file, then
# as directory, then fall back to displaying a 404.
try_files $uri $uri/ /index.php?$query_string;
}

location @ws {
proxy_pass http://127.0.0.1:6001;
proxy_set_header Host $host;
proxy_read_timeout 60;
proxy_connect_timeout 60;
proxy_redirect off;

# Allow the use of websockets
proxy_http_version 1.1;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection 'upgrade';
proxy_set_header Host $host;
proxy_cache_bypass $http_upgrade;
# proxy_set_header X-Real-IP $remote_addr;
# proxy_set_header X-Forward-For $proxy_add_x_forwarded_for;
}

# pass PHP scripts to FastCGI server
#
location ~ \.php$ {
include snippets/fastcgi-php.conf;
#
# # With php-fpm (or other unix sockets):
fastcgi_pass unix:/var/run/php/php7.3-fpm.sock;
# # With php-cgi (or other tcp sockets):
# fastcgi_pass 127.0.0.1:9000;
}

# deny access to .htaccess files, if Apache's document root
# concurs with nginx's one
#
location ~ /\.ht {
deny all;
}

listen [::]:443 ssl ipv6only=on; # managed by Certbot
listen 443 ssl; # managed by Certbot
ssl_certificate /etc/letsencrypt/live/profileapp.org/fullchain.pem; # managed by Certbot
ssl_certificate_key /etc/letsencrypt/live/profileapp.org/privkey.pem; # managed by Certbot
include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}

server {
if ($host = www.profileapp.org) {
return 301 https://$host$request_uri;
} # managed by Certbot

if ($host = profileapp.org) {
return 301 https://$host$request_uri;
} # managed by Certbot

listen 80 default_server;
listen [::]:80 default_server;

server_name profileapp.org www.profileapp.org;
return 404; # managed by Certbot

}

<------------ END FILE ----------->

Client Configuration: $ /var/www/laravel/resources/js/bootstrap.js

...

import Echo from 'laravel-echo';

window.Pusher = require('pusher-js');

window.Echo = new Echo({
broadcaster: 'pusher',
key: 'appkey1',
// cluster: 'us2',
encrypted: true,
wsHost: window.location.hostname,
wsPort: 6001,
wssPort: 6001,
disableStats: true,
enabledTransports: ["ws", "wss"],
});

<--------- END FILE ---------->

SSL Settings: $ /var/www/laravel/config/websockets.php

'ssl' => [

'local_cert' => '/etc/letsencrypt/live/profileapp.org/fullchain.pem',

'local_pk' => '/etc/letsencrypt/live/profileapp.org/privkey.pem',

'passphrase' => null,

'allow_self_signed' => true,

'verify_peer' => false,
],

<--------- END FILE ----------->

Pusher Configuration: $ /var/www/laravel/config/broadcasting.php

'pusher' => [
'driver' => 'pusher',
'key' => env('PUSHER_APP_KEY'),
'secret' => env('PUSHER_APP_SECRET'),
'app_id' => env('PUSHER_APP_ID'),
'options' => [
'cluster' => env('PUSHER_APP_CLUSTER'),
'encrypted' => true,  
'host' => '127.0.0.1',  // 'host' => 'https://profileapp.org'
'port' => 6001,
'scheme' => 'https'
],

<-------- END FILE -------------->

**Comment by francislavoie** [comment]
As far as I can tell, it's because you didn't open port 6001 on your server.

**Comment by anwarhamoude** [comment]
ufw allow 6001/tcp

I had disabled the firewall yesterday to eliminate the possibility it was that. Maybe I poked at the server so much it is garbage? I thought I was good at programming servers. I'm setting this application up on Forge/DigitalOcean for a trial run. The application is just a test anyway- I wanted to see those comments zap in there. I really have tried every screwball variation I found.

**Comment by anwarhamoude** [comment]
I did try every possible combination.
In the final working configuration I am using the domain name in Pusher.
'host' => 'https://profileapp.org'

The problem was supervisord. Strangely the command sudo supervisorctl status showed active. When I was trying this project on Forge I had run netstat and saw port 6001 being used. The reason was Forge has some issues running supervisord, it wasn't doing it. But no it did it and the application tested. I poked at this server some more. Seeing that the no process was running on port 6001 I came to the realization the websocket server simply wasn't running.
